ui/src-tauri/src/main.rs

```diff
@@ -107,11 +133,15 @@ impl Authorizer for User {
     fn setup<'s>(&'s mut self, setup: &'s Setup) -> UnitFuture<'s> {
         let window = self.window.clone();
         Box::pin(async move {
-            // NOTE: We have to wait here until the UI listener is registered.
-            sleep(Duration::from_millis(500)).await;
-            window
-                .emit("connect", setup)
-                .expect("The `connect` command failed to be emitted to the window.");
+            ui_ready!(
+                {
+                    window
+                        .emit("connect", setup)
+                        .expect("The `connect` command failed to be emitted to the window.");
+                },
+                10000,
+                { panic!() }
```

Comment:
nit: might be good to put a message in the panic

Reply:
Yep, added it